example/dummy/dummy.go

```diff
@@ -40,28 +43,45 @@ func (app *DummyApplication) CheckTx(tx []byte) types.Result {
 }
 
 func (app *DummyApplication) Commit() types.Result {
-	hash := app.state.Hash()
+	// Save a new version
+	var hash []byte
+	var err error
+
+	if app.state.Size() > 0 {
```

Comment:
do we need this condition? I mean, by the time we commit the size should always be > 0, no?

Reply:
Well. We have tests that commit a few empty blocks before adding a TX. They were returning errors here, which broke things. So I added tests. Thus the slack ping to alexis

Reply:
Hm. For the persistent dummy I ended up adding the constraint that InitChain must be called, which will make the state non-zero. In general would be nice if we could still get versions of empty trees so we dont have to worry about this edge condition

Reply:
We can use `!app.state.IsEmpty()` now which is a bit cleaner, but if a tree has no root, what are we saving exactly? We could potentially have it just save a version number but hmmm..

Reply:
It's just for useability. The tree is still something the app thinks about and has a root (even if empty). And for recover logic, we like the tree version to equal the height, even if the tree hasn't changed.